src/AVFoundation/AVAudioFormat.cs

```diff
@@ -25,7 +27,7 @@ public partial class AVAudioFormat {
 				return true;
 			if ((object) a == null ^ (object) b == null)
 				return false;
-			return a.Equals (b);
+			return a!.Equals (b);
```

Comment:
Do we really need that !you should be checking again null and get that warning out. Move the == to use is

Reply:
The compiler complained here when using `is null`. I think it may have had something to do with the `^`

I decided to use the exclamation point since we first check if they are equal, then we check if one of the two are null, which I believe hits all the possibilities of it being null? @mandel-macaque

Reply:
@mandel-macaque  Edit: I can do this: `if (a is null ^ b is null)` and remove the object cast
When it had the (object) cast, it gave this warning/erorr: "Converting null literal or possible null value to non-nullable type."

What would an appropriate null check for this return if the objects are null and not null?

Reply:
Changing the previous condition to: `if (a is null ^ b is null)` (or `if (a is null || b is null)`) looks correct.